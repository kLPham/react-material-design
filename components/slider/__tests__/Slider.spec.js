import React from 'react';
import { shallow } from 'enzyme';
import Slider from '../index';

describe('<Slider />', () => {
    it('should render', () => {
        const slider = shallow(
            <Slider />,
    );
        expect(slider).toMatchSnapshot();
    });

    it('shound render with initialValue', () => {
        const initialValue = 50;
        const select = shallow(
            <Slider initialValue={initialValue} />,
      );
        expect(select).toMatchSnapshot();
    });

    it('shound render disabled', () => {
        const disabled = true;
        const select = shallow(
            <Slider disabled={disabled} />,
    );
        expect(select).toMatchSnapshot();
    });
});
