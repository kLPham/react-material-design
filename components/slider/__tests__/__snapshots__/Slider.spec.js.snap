// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> should render 1`] = `
<div
  aria-label="Select Value"
  aria-valuemax="100"
  aria-valuemin="0"
  className="mdc-slider"
  role="slider"
  tabIndex="0"
>
  <div
    className="mdc-slider__track-container"
  >
    <div
      className="mdc-slider__track"
    />
  </div>
  <div
    className="mdc-slider__thumb-container"
  >
    <svg
      className="mdc-slider__thumb"
      height="21"
      width="21"
    >
      <circle
        cx="10.5"
        cy="10.5"
        r="7.875"
      />
    </svg>
    <div
      className="mdc-slider__focus-ring"
    />
  </div>
</div>
`;

exports[`<Slider /> shound render disabled 1`] = `
<div
  aria-disabled={true}
  aria-label="Select Value"
  aria-valuemax="100"
  aria-valuemin="0"
  className="mdc-slider"
  role="slider"
  tabIndex="0"
>
  <div
    className="mdc-slider__track-container"
  >
    <div
      className="mdc-slider__track"
    />
  </div>
  <div
    className="mdc-slider__thumb-container"
  >
    <svg
      className="mdc-slider__thumb"
      height="21"
      width="21"
    >
      <circle
        cx="10.5"
        cy="10.5"
        r="7.875"
      />
    </svg>
    <div
      className="mdc-slider__focus-ring"
    />
  </div>
</div>
`;

exports[`<Slider /> shound render with initialValue 1`] = `
<div
  aria-label="Select Value"
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow={50}
  className="mdc-slider"
  role="slider"
  tabIndex="0"
>
  <div
    className="mdc-slider__track-container"
  >
    <div
      className="mdc-slider__track"
    />
  </div>
  <div
    className="mdc-slider__thumb-container"
  >
    <svg
      className="mdc-slider__thumb"
      height="21"
      width="21"
    >
      <circle
        cx="10.5"
        cy="10.5"
        r="7.875"
      />
    </svg>
    <div
      className="mdc-slider__focus-ring"
    />
  </div>
</div>
`;
